### ISSUE TITLE
audio_encoder missing from pretrained SAVi model

### ISSUE THREAD

**Issue by auroracramer** [facebookresearch/sound-spaces#75]
While inspecting the pretrained weights for the SAVi model, I noticed that there's no `audio_encoder` model in the policy models `state_dict`:

```
weights_path = './data/pretrained_weights/semantic_audionav/savi/best_val.pth'
loaded_weights = torch.load(weights_path, map_location=torch.device('cpu'))
[k for k in loaded_weights['state_dict'].keys() if 'audio_encoder' in k] # returns empty list []
```

Is this intentional? According to Section 4.1 in the Semantic Audio-Visual Navigation paper, it seems like the SAVi model should have an audio encoder separate from the one used for the goal encoder. However, I also noticed that the default configs for SAVi don't include spectrograms in the SENSOR observations (https://github.com/facebookresearch/sound-spaces/blob/main/ss_baselines/savi/config/semantic_audionav/savi.yaml), while the available task observations do include spectrograms (https://github.com/facebookresearch/sound-spaces/blob/main/configs/semantic_audionav/savi/mp3d/semantic_audiogoal.yaml). However, I'm still getting familiar with the code so it's entirely possible I'm misunderstanding something. Is there something that I'm missing?

Also, is there a SAVi model with publicly available pretrained weights that includes this audio observation encoder?

**Comment by ChanganVR** [comment]
Hi @auroracramer, sorry about the terrible delay. I hope you have figured it out already. If not, see my response below.

The audio encoder is part of the actor_critic network in the pretrained weights and the label is `actor_critic.net.goal_encoder` while visual encoder is `actor_critic.net.visual_encoder`. So the pretrained weight does include the audio encoder already.

The configuration of the Habitat/SoundSpaces system is hierarchical. The config you shared is training config and task config is here: https://github.com/facebookresearch/sound-spaces/blob/main/configs/semantic_audionav/savi/mp3d/semantic_audiogoal.yaml, which uses spectrogram sensor.

Let me know if you have other questions.

**Comment by auroracramer** [comment]
@ChanganVR no worries, thanks for the response. :)

Is there no observation encoder for audio? From Section 4.1 and Figure 2 in the paper, it looks like there should be two audio encoders, one for the observation encoder and another for the goal descriptor (corresponding to f_B and f_D, respectively). Is this not actually the case?
